File: python/04_014.py
class Computer:
    def __init__(self, builder):
        # Build computer with specifications
        self.cpu = builder.cpu
        self.ram = builder.ram

class ComputerBuilder:
    def __init__(self):
        self.cpu = None
        self.ram = None
    
    def set_cpu(self, cpu):
        self.cpu = cpu
        return self
    
    def set_ram(self, ram):
        self.ram = ram
        return self
    
    def build(self):
        return Computer(self)

File: java/04_014.java
public class Computer {
    private Computer(Builder builder) {
        // Build computer with specifications
    }
    
    public static class Builder {
        private String cpu;
        private String ram;
        
        public Builder setCPU(String cpu) {
            this.cpu = cpu;
            return this;
        }
        
        public Computer build() {
            return new Computer(this);
        }
    }
}